src/components/structures/TagPanel.js

```diff
@@ -115,8 +53,21 @@ export default React.createClass({
                 selectedTags: FilterStore.getSelectedTags(),
             });
         });
+        this._tagOrderStoreToken = TagOrderStore.addListener(() => {
+            if (this.unmounted) {
+                return;
+            }
 
-        this._fetchJoinedRooms();
+            const orderedTags = TagOrderStore.getOrderedTags() || TagOrderStore.getAllTags();
```

Comment:
It seems a bit strange to me that the view would ever have to worry about having to fall back to another piece of data if one of these is missing.

Reply:
Yep, true let's make sure the store exposes only the state that should be visible.